spacy_udpipe/udpipe.py

```diff
@@ -0,0 +1,202 @@
+import re
+from typing import Dict, List, Optional, Union
+
+from ufal.udpipe import (InputFormat, Model, OutputFormat, ProcessingError,
+                         Sentence)
+
+from .utils import get_path
+
+
+class PretokenizedInputFormat(object):
+    """Dummy tokenizer for pretokenized input.
+
+    Execution speed might be slow compared to other UDPipe tokenizers
+    due to pure Python implementation. Mocks InputFormat API to enable
+    plug-and-play behaviour.
+    """
+    NO_SPACE = "SpaceAfter=No"
+
+    def setText(self, text: str) -> None:
+        """Store text in iterable lines for tokenization.
+
+        text: string, where each sentence is on a line and tokens
+              are separated by tabs.
+        """
+        self.lines = iter(text.split("\n"))
+
+    def nextSentence(self, sentence: Sentence, _: ProcessingError) -> bool:
+        """Tokenize each line from stored lines and store tokens in sentence.
+
+        sentence: UDPipe container for storing tokens.
+        """
+        try:
+            line = next(self.lines)
+        except StopIteration:
+            return False
+        tokens = line.split("\t") + [str(None)]  # EOS token
+        for i, (token, next_token) in enumerate(zip(tokens[:-1], tokens[1:])):
```

Comment:
nitpick: this looks like a nice functional solution, but I'd prefer a simpler one using `i` to check token[i + 1] as I would guess it is faster. Also you have word.id commented out.

Reply:
This is more Pythonic according to [Ch 1, Item 8](https://effectivepython.com/). Why would it be faster with explicit indexing?

Reply:
Can't say I tested it, but it just seems like less operations are involved with explicit indexing (I'm not saying the difference is really big). I'm just thinking that indexing list elements is faster then creating two iterable objects. 

I don't agree that your case here suits what you reference. It names two iterables. Here you have a one list (which you know the length of) which you then use as two iterables.